src/wp-includes/robots-template.php

```diff
@@ -0,0 +1,171 @@
+ * tell web robots not to index the page content. Add this to the
+ * {@see 'wp_robots'} filter.
+ *
+ * Typical usage is as a {@see 'wp_robots'} callback:
+ *
+ *     add_filter( 'wp_robots', 'wp_robots_noindex' );
+ *
+ * @since 5.6.0
+ * @see wp_robots_no_robots()
+ *
+ * @param array $robots Associative array of robots directives.
+ * @return array Filtered robots directives.
+ */
+function wp_robots_noindex( array $robots ) {
+	if ( ! get_option( 'blog_public' ) ) {
+		return wp_robots_no_robots( $robots );
+	}
+
+	return $robots;
+}
+
+/**
+ * Adds noindex to the robots meta tag.
+ *
+ * This directive tells web robots not to index the page content.
+ *
+ * Typical usage is as a {@see 'wp_robots'} callback:
+ *
+ *     add_filter( 'wp_robots', 'wp_robots_no_robots' );
+ *
+ * @since 5.6.0
+ *
+ * @param array $robots Associative array of robots directives.
+ * @return array Filtered robots directives.
+ */
+function wp_robots_no_robots( array $robots ) {
+	$robots['noindex'] = true;
+
+	if ( get_option( 'blog_public' ) ) {
+		$robots['follow'] = true;
+	} else {
+		$robots['nofollow'] = true;
+	}
+
+	return $robots;
+}
+
+/**
+ * Adds noindex and noarchive to the robots meta tag.
+ *
+ * This directive tells web robots not to index or archive the page content and
+ * is recommended to be used for sensitive pages.
+ *
+ * Typical usage is as a {@see 'wp_robots'} callback:
+ *
+ *     add_filter( 'wp_robots', 'wp_robots_sensitive_page' );
+ *
+ * @since 5.6.0
+ *
+ * @param array $robots Associative array of robots directives.
+ * @return array Filtered robots directives.
+ */
+function wp_robots_sensitive_page( array $robots ) {
+	$robots['noindex']   = true;
+	$robots['noarchive'] = true;
+	return $robots;
+}
+
+/**
+ * Adds 'max-image-preview:large' to the robots meta tag if required by the
+ * site configuration.
+ *
+ * Typical usage is as a {@see 'wp_robots'} callback:
+ *
+ *     add_filter( 'wp_robots', 'wp_robots_media_search_engine_visibility' );
+ *
+ * @since 5.6.0
+ * @see wp_robots_max_image_preview()
+ *
+ * @param array $robots Associative array of robots directives.
+ * @return array Filtered robots directives.
+ */
+function wp_robots_media_search_engine_visibility( array $robots ) {
+	if ( get_option( 'blog_public' ) && get_option( 'media_search_engine_visibility' ) ) {
+		return wp_robots_max_image_preview( $robots );
+	}
+
+	return $robots;
+}
+
+/**
+ * Adds 'max-image-preview:large' to the robots meta tag.
+ *
+ * This directive tells web robots that large image previews are allowed to be
+ * displayed, e.g. in search engines.
+ *
+ * Typical usage is as a {@see 'wp_robots'} callback:
+ *
+ *     add_filter( 'wp_robots', 'wp_robots_max_image_preview' );
+ *
+ * @since 5.6.0
+ *
+ * @param array $robots Associative array of robots directives.
+ * @return array Filtered robots directives.
+ */
+function wp_robots_max_image_preview( array $robots ) {
```

Comment:
If someone filters `wp_robots` to say `__return_null` then the `array` type would cause a fatal error, no?

Reply:
I'm not sure we need to cater for that as that would clearly be an incorrect usage of the filter, which should cause a failure during development.

Reply:
We also have `__return_empty_array` if someone wants a shortcut to empty everything out.

Reply:
I'm just saying, you never know what a WP dev will do.